bokeh/document/document.py

```diff
@@ -132,6 +132,17 @@ def session_callbacks(self):
         '''
         return list(self._session_callbacks)
 
+    @property
+    def session_destroyed_callbacks(self):
+        ''' A list of all the on_session_destroyed callbacks on this document.
+
+        '''
+        return list(self._session_destroyed_callbacks)
```

Comment:
Why the list asymmetry here? If you want to return a copy, why not call `dict` and return that?

Reply:
Yeah, no good reason, I was copying `session_callbacks` but that is actually stored as a set.

Reply:
I think setting it to `{}` above is a mistake too, then, you'd need to assign it `set()` I think it should be consistent throughout, whatever it is

Reply:
Yes, I'll store these callbacks in a set too.